Homomorphic encryption libraries: Palisade, HELib, SEAL, and NTL.

wohlbier opened the pull request:
Add packages for https://github.com/homenc/HElib and it's dependency https://github.com/libntl/ntl.

sei-jgwohlbier committed:
working version of helib package and ntl dependency.

sei-jgwohlbier committed:
working version of helib package and ntl dependency.

sei-jgwohlbier committed:
Fix style errors.

sei-jgwohlbier committed:
Fix style errors.

lucbv committed:
Kokkos Kernels: updating maintainers and releases (#26855)

Adding the latests releases of Kokkos Kernels and the official maintainers for the package.

sei-jgwohlbier committed:
Remove boiler plate comments.

sei-jgwohlbier committed:
Remove boiler plate comments.

adamjstewart committed:
py-numpy: add v1.21.3 (#26860)

shuds13 committed:
libEnsemble: add v0.8.0 (#26864)

downloadico committed:
tramonto: add new package (#26858)

alalazo commented on var/spack/repos/builtin/packages/helib/package.py:
```diff
@@ -0,0 +1,53 @@
+    faster, focusing mostly on effective use of the Smart-Vercauteren
+    ciphertext packing techniques and the Gentry-Halevi-Smart optimizations.
+    See this report for a description of a few of the algorithms using in this
+    library."""
```
Can you have a short first paragraph, followed by a longer description later (disclaimer, I didn't check the snippet below for style) e.g.
```suggestion
    """HElib is an open-source (Apache License v2.0) software library that
    implements homomorphic encryption (HE). 
    
    Currently available schemes are the implementations of the 
    Brakerski-Gentry-Vaikuntanathan (BGV) scheme
    with bootstrapping and the Approximate Number scheme of Cheon-Kim-Kim-Song
    (CKKS), along with many optimizations to make homomorphic evaluation run
    faster, focusing mostly on effective use of the Smart-Vercauteren
    ciphertext packing techniques and the Gentry-Halevi-Smart optimizations.
    See this report for a description of a few of the algorithms using in this
    library."""
```
Spack will use the short paragraph in contexts where a short description is needed, and the full docstring where more info needs to be provided to users.

alalazo commented on var/spack/repos/builtin/packages/helib/package.py:
```diff
@@ -0,0 +1,53 @@
+    (CKKS), along with many optimizations to make homomorphic evaluation run
+    faster, focusing mostly on effective use of the Smart-Vercauteren
+    ciphertext packing techniques and the Gentry-Halevi-Smart optimizations.
+    See this report for a description of a few of the algorithms using in this
```
This last sentence can be removed since we don't have any handle to "this" report.

wohlbier replied:
Making suggested change.

alalazo commented on var/spack/repos/builtin/packages/helib/package.py:
```diff
@@ -0,0 +1,53 @@
+    homepage = "https://github.com/homenc/HElib"
+    url      = "https://github.com/homenc/HElib/archive/refs/tags/v2.2.1.tar.gz"
+
+    # notify when the package is updated.
```
```suggestion
```

alalazo commented on var/spack/repos/builtin/packages/helib/package.py:
```diff
@@ -0,0 +1,53 @@
+    version('1.0.2',        sha256='b907eaa8381af3d001d7fb8383273f4c652415b3320c11d5be2ad8f19757c998')
+
+    variant('shared', values=bool, default=False,
+            description='Build shared library.')
```
```suggestion
    variant('shared', default=False, description='Build shared library.')
```

alalazo commented on var/spack/repos/builtin/packages/helib/package.py:
```diff
@@ -0,0 +1,53 @@
+            '-DNTL_DIR={0}'.format(spec['ntl'].prefix)
+        ]
+        if '+shared' in spec:
+            args.append('-DBUILD_SHARED=ON')
```
Can you try:
```suggestion
        args = [
            self.define('ENABLE_TEST', 'ON'),
            self.define('GMP_DIR', spec['gmp'].prefix),
            self.define('NTL_DIR', spec['ntl'].prefix),
            self.define_from_variant('BUILD_SHARED', 'shared')
        ]
```
? This is the idiomatic way to define properties for CMake.

alalazo commented on var/spack/repos/builtin/packages/ntl/package.py:
```diff
@@ -0,0 +1,73 @@
+    homepage = "https://libntl.org"
+    url      = "https://github.com/libntl/ntl/archive/refs/tags/v11.5.1.tar.gz"
+
+    # notify when the package is updated.
```
```suggestion
```

alalazo commented on var/spack/repos/builtin/packages/ntl/package.py:
```diff
@@ -0,0 +1,73 @@
+    NTL  -- a library for doing numbery theory --  version 11.5.1
+    Release date: 2021.06.23
+
+    Author: Victor Shoup (victor@shoup.net)
```
Would it be possible to use:
```suggestion
    NTL  -- a library for doing numbery theory
```
? The information removed is subject to more frequent changes and should be found in the homepage of the package that is reported below.

alalazo commented on var/spack/repos/builtin/packages/ntl/package.py:
```diff
@@ -0,0 +1,73 @@
+    * doc/tour-unix.html for unix systems
+    * doc/tour-win.html for Windows systems
+
+    The latest version of NTL is available at http://www.shoup.net.
```
Can we also remove this:
```suggestion
```
?

alalazo commented on var/spack/repos/builtin/packages/ntl/package.py:
```diff
@@ -0,0 +1,73 @@
+    version('11.4.4', sha256='2ce7a10fadbed6c3859d72c859612a4ca0dbdf6a9db99db4261422b7f0804bfa')
+
+    variant('shared', values=bool, default=False,
+            description='Build shared library.')
```
```suggestion
    variant('shared', default=False, description='Build shared library.')
```

alalazo commented on var/spack/repos/builtin/packages/ntl/package.py:
```diff
@@ -0,0 +1,73 @@
+    def build(self, spec, prefix):
+        with working_dir('src'):
+            make()
+            # make('check')
```
Can you either remove this or leave in the code an explanation why it was commented out?

wohlbier replied:
Will make suggested change.

alalazo reviewed (changes requested):
Thanks for the contribution. I left a few comments, let me know what you think.

RichardABunt committed:
arm-forge: add v21.1 (#26852)

edwardhartnett committed:
bacio: add v2.5.0 (#26851)

ajaust committed:
Make superlu-dist@7.1.0 request CMake 3.18.1 or newer (#26849)

wdconinc committed:
kassiopeia: add v3.8.0; new variants log4cxx, boost (#26841)

vsoch committed:
libabigail: support source install (#26807)

Signed-off-by: vsoch <vsoch@users.noreply.github.com>
Co-authored-by: Massimiliano Culpo <massimiliano.culpo@gmail.com>

brugger1 committed:
vtk: make use of system GLEW dependent on osmesa being disabled. (#26764)

haampie committed:
Make 'spack location -e' print the current env, and 'spack cd -e' go to the current env (#26654)

ajaust committed:
[py-pyprecice] add v2.3.0.1 and fix hash of v2.2.1.1 (#26846)

eugeneswalker committed:
update E4S CI environments in preparation for 21.11 release (#26826)

* update E4S CI environments in preparation for 21.11 release

* e4s ci env: use clingo

wohlbier committed:
Update var/spack/repos/builtin/packages/helib/package.py

Co-authored-by: Massimiliano Culpo <massimiliano.culpo@gmail.com>

wohlbier committed:
Update var/spack/repos/builtin/packages/helib/package.py

Co-authored-by: Massimiliano Culpo <massimiliano.culpo@gmail.com>

wohlbier committed:
Update var/spack/repos/builtin/packages/helib/package.py

Co-authored-by: Massimiliano Culpo <massimiliano.culpo@gmail.com>

wohlbier committed:
Update var/spack/repos/builtin/packages/helib/package.py

Co-authored-by: Massimiliano Culpo <massimiliano.culpo@gmail.com>

wohlbier committed:
Update var/spack/repos/builtin/packages/helib/package.py

Co-authored-by: Massimiliano Culpo <massimiliano.culpo@gmail.com>

wohlbier committed:
Update var/spack/repos/builtin/packages/helib/package.py

Co-authored-by: Massimiliano Culpo <massimiliano.culpo@gmail.com>

wohlbier committed:
Update var/spack/repos/builtin/packages/helib/package.py

Co-authored-by: Massimiliano Culpo <massimiliano.culpo@gmail.com>

wohlbier committed:
Update var/spack/repos/builtin/packages/helib/package.py

Co-authored-by: Massimiliano Culpo <massimiliano.culpo@gmail.com>

wohlbier committed:
Update var/spack/repos/builtin/packages/ntl/package.py

Co-authored-by: Massimiliano Culpo <massimiliano.culpo@gmail.com>

wohlbier committed:
Update var/spack/repos/builtin/packages/ntl/package.py

Co-authored-by: Massimiliano Culpo <massimiliano.culpo@gmail.com>

wohlbier committed:
Update var/spack/repos/builtin/packages/ntl/package.py

Co-authored-by: Massimiliano Culpo <massimiliano.culpo@gmail.com>

wohlbier committed:
Update var/spack/repos/builtin/packages/ntl/package.py

Co-authored-by: Massimiliano Culpo <massimiliano.culpo@gmail.com>

wohlbier committed:
Update var/spack/repos/builtin/packages/ntl/package.py

Co-authored-by: Massimiliano Culpo <massimiliano.culpo@gmail.com>

wohlbier committed:
Update var/spack/repos/builtin/packages/ntl/package.py

Co-authored-by: Massimiliano Culpo <massimiliano.culpo@gmail.com>

wohlbier committed:
Update var/spack/repos/builtin/packages/ntl/package.py

Co-authored-by: Massimiliano Culpo <massimiliano.culpo@gmail.com>

wohlbier committed:
Update var/spack/repos/builtin/packages/ntl/package.py

Co-authored-by: Massimiliano Culpo <massimiliano.culpo@gmail.com>

sei-jgwohlbier committed:
make requested change from review.

sei-jgwohlbier committed:
make requested change from review.

sei-jgwohlbier committed:
Merge branch 'helib' of github.com:wohlbier/spack into helib

sei-jgwohlbier committed:
make requested change from review.

sei-jgwohlbier committed:
make requested change from review.

manuelakuhn committed:
py-neurokit2: add 0.1.4.1 (#26871)

* py-neurokit2: add 0.1.4.1

* Update var/spack/repos/builtin/packages/py-neurokit2/package.py

Co-authored-by: Adam J. Stewart <ajstewart426@gmail.com>

Co-authored-by: Adam J. Stewart <ajstewart426@gmail.com>

rc committed:
py-sfepy: add v2021.3 (#26865)

frankwillmore committed:
Add vacuumms package (#26838)

CamStan committed:
scr/veloc: component releases (#26716)

* scr/veloc: component releases

Update the ECP-VeloC component packages in preparation for an
upcoming scr@3.0rc2 release.

All
- Add new release versions
- Add new `shared` variant for all components
- Add zlib link dependency to packages that were missing it
- Add maintainers
- Use self.define and self.define_from_variant to clean up cmake_args

axl
- Add independent vendor async support variants

rankstr
- Update older version sha that fails checksum on install

* Fix scr build error

Lock dependencies for scr@3.0rc1 to the versions released at the same
time.

mkstoyanov committed:
heffte: new version with rocm (#26878)

douglasjacobsen committed:
Add GCS Bucket Mirrors (#26382)

This commit contains changes to support Google Cloud Storage 
buckets as mirrors, meant for hosting Spack build-caches. This 
feature is beneficial for folks that are running infrastructure on 
Google Cloud Platform. On public cloud systems, resources are
ephemeral and in many cases, installing compilers, MPI flavors,
and user packages from scratch takes up considerable time.

Giving users the ability to host a Spack mirror that can store build
caches in GCS buckets offers a clean solution for reducing
application rebuilds for Google Cloud infrastructure.

Co-authored-by: Joe Schoonover <joe@fluidnumerics.com>

haampie committed:
Remove hard-coded repository in centos6 unit test (#26885)

This makes it easier to run tests in a fork.

haampie committed:
Backport #186 from py-py to fix macOS failures (#26653)

Backports the relevant bits of https://github.com/pytest-dev/py/commit/0f77b6e66f79c07dbb657e2b4a7e94263eacc01b

sei-jgwohlbier committed:
Set CXX sent to configure to be the spack c++ compiler.

sei-jgwohlbier committed:
Set CXX sent to configure to be the spack c++ compiler.

haampie committed:
Shorten long shebangs only if the execute permission is set (#26899)

The OS should only interpret shebangs, if a file is executable. 

Thus, there should be no need to modify files where no execute bit is set. 

This solves issues that are e.g. encountered while packaging software as 
COVISE (https://github.com/hlrs-vis/covise), which includes example data
in Tecplot format. The sbang post-install hook is applied to every installed
file that starts with the two characters #!, but this fails on the binary Tecplot
files, as they happen to start with #!TDV. Decoding them with UTF-8 fails 
and an exception is thrown during post_install.

Co-authored-by: Martin Aumüller <aumuell@reserv.at>

manuelakuhn committed:
py-jupyterlab: add 3.1.18 and 3.2.1 (#26896)

iarspider committed:
Add py-asn1crypto@1.4.0 (#26895)

rc committed:
py-sfepy: fix version 2021.3 tar.gz hash (#26890)

wdconinc committed:
acts: add v14.0.0, v14.1.0 added "python" and "analysis" variants (#26657)

iarspider committed:
Use pypi for astroid; add versions 2.7.3 and 2.8.3; update dependencies (#26900)

* Use pypi for astroid; add versions 2.7.3 and 2.8.3; update dependencies

* py-six is not required for recent versions

iarspider committed:
Update py-aiohttp to 3.7.4 and py-chartdet to 4.0 (#26893)

* Update py-aiohttp to 3.7.4 and py-chartdet to 4.0

* Changes from review

* Update package.py

* Update var/spack/repos/builtin/packages/py-chardet/package.py

Co-authored-by: Adam J. Stewart <ajstewart426@gmail.com>

Co-authored-by: Adam J. Stewart <ajstewart426@gmail.com>

haampie committed:
Document backport in py (#26897)

wdconinc committed:
acts: fix misspelled variant in depends_on (#26904)

Correct variant is plural `unit_tests`.

sei-jgwohlbier committed:
Add palisade HE library as well.

sei-jgwohlbier committed:
Merge branch 'helib' of github.com:wohlbier/spack into helib

sei-jgwohlbier committed:
fix style error.

sei-jgwohlbier committed:
fix style error

wohlbier commented:
Hi, is there something that I need to do to address `codecov/patch` and `codecov/project`?

Also, I made the mistake of rebasing my feature branch between pushes. Is this Ok, or do I need to clean things up. Thanks.

jgw

sei-jgwohlbier committed:
add a package for microsoft seal.

wohlbier commented:
I'm going to close this and do a new PR that has one commit on top of the tip of develop.
